kiali/kiali
# Workloads GVK support + Custom Workloads

### Describe the change

Workloads were tight to their `Kind`, which was making impossible to read those ones which are Custom and has the same Kind but different 'apiVersion'.
Now Workloads support GroupVersionKind  instead of 'type' field.

### Steps to test the PR
Install Kruise, wait until it is up.
`helm install kruise openkruise/kruise --version 1.7.2`
Install this custom DaemonSet.
```
apiVersion: apps.kruise.io/v1alpha1
kind: DaemonSet
metadata:
  name: fluentd-elasticsearch
  labels:
    k8s-app: fluentd-logging
spec:
  selector:
    matchLabels:
      name: fluentd-elasticsearch
  updateStrategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 1
  template:
    metadata:
      labels:
        name: fluentd-elasticsearch
    spec:
      tolerations:
      # these tolerations are to have the daemonset runnable on control plane nodes
      # remove them if your control plane nodes should not run pods
      - key: node-role.kubernetes.io/control-plane
        operator: Exists
        effect: NoSchedule
      - key: node-role.kubernetes.io/master
        operator: Exists
        effect: NoSchedule
      containers:
      - name: fluentd-elasticsearch
        image: quay.io/fluentd_elasticsearch/fluentd:v2.5.2
        resources:
          limits:
            memory: 200Mi
          requests:
            cpu: 100m
            memory: 200Mi
        volumeMounts:
        - name: varlog
          mountPath: /var/log
        - name: varlibdockercontainers
          mountPath: /var/lib/docker/containers
          readOnly: true
      terminationGracePeriodSeconds: 30
      volumes:
      - name: varlog
        hostPath:
          path: /var/log
      - name: varlibdockercontainers
        hostPath:
          path: /var/lib/docker/containers
```

List Workloads, the newly added DaemonSet should be listed.
Click on it.
Details page comes with disabled Action:
![Screenshot from 2024-11-15 11-33-32](https://github.com/user-attachments/assets/742414a0-b1c1-47fd-a0b9-6abfb67f03b2)


### Automation testing

Existing tests modified to support GVK.

### Issue reference
https://github.com/kiali/kiali/issues/7820
